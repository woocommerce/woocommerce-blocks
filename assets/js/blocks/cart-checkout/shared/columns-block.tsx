/**
 * External dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';

export const Columns = ( {
	children,
	...props
}: {
	children?: React.ReactNode;
} ): JSX.Element => {
	const blockProps = useBlockProps( props );

	return <div { ...blockProps }>{ children }</div>;
};
