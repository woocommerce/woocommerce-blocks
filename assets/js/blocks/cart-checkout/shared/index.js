export * from './hacks';
export * from './use-forced-layout';
export * from './editor-utils';
export * from './use-view-switcher';
