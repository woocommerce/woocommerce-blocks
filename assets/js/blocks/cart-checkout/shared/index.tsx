export * from './hacks';
export * from './editor-utils';
export * from './use-forced-layout';
