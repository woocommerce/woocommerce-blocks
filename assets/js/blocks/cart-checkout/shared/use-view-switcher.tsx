/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { useState } from '@wordpress/element';
import { useDispatch, select } from '@wordpress/data';
import { ToolbarGroup, ToolbarDropdownMenu } from '@wordpress/components';
import { Icon, eye } from '@woocommerce/icons';
import { store as blockEditorStore } from '@wordpress/block-editor';

interface View {
	view: string;
	label: string;
	icon: string | JSX.Element;
}

export const useViewSwitcher = (
	clientId: string,
	views: View[]
): {
	currentView: string;
	component: JSX.Element;
} => {
	const initialView = views[ 0 ];
	const [ currentView, setCurrentView ] = useState( initialView );
	const { selectBlock } = useDispatch( 'core/block-editor' );
	const { getBlock } = select( blockEditorStore );

	const ViewSwitcherComponent = (
		<ToolbarGroup>
			<ToolbarDropdownMenu
				label={ __( 'Switch view', 'woo-gutenberg-products-block' ) }
				text={ currentView.label }
				icon={
					<Icon srcElement={ eye } style={ { marginRight: '8px' } } />
				}
				controls={ views.map( ( view ) => ( {
					...view,
					title: <span>{ view.label }</span>,
					isActive: view.view === currentView.view,
					onClick: () => {
						setCurrentView( view );
						selectBlock(
							getBlock( clientId ).innerBlocks.find(
								( block: { name: string } ) =>
									block.name === view.view
							)?.clientId || clientId
						);
					},
				} ) ) }
			/>
		</ToolbarGroup>
	);

	return {
		currentView: currentView.view,
		component: ViewSwitcherComponent,
	};
};
