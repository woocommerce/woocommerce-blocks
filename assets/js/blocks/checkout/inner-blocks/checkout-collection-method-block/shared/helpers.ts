/**
 * External dependencies
 */
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

/**
 * Returns the cheapest rate that isn't a local pickup.
 *
 * @param {Array|undefined} shippingRates Array of shipping Rate.
 *
 * @return {Object|undefined} cheapest rate.
 */
export function getShippingStartingPrice(
	shippingRates: CartShippingPackageShippingRate[]
): CartShippingPackageShippingRate | undefined {
	if ( shippingRates ) {
		return shippingRates.reduce(
			(
				lowestRate: CartShippingPackageShippingRate | undefined,
				currentRate: CartShippingPackageShippingRate
			) => {
				if ( currentRate.method_id === 'local_pickup' ) {
					return lowestRate;
				}
				if (
					lowestRate === undefined ||
					currentRate.price < lowestRate.price
				) {
					return currentRate;
				}
				return lowestRate;
			},
			undefined
		);
	}
}

/**
 * Returns the cheapest rate that is a local pickup.
 *
 * @param {Array|undefined} shippingRates Array of shipping Rate.
 *
 * @return {Object|undefined} cheapest rate.
 */
export function getLocalPickupStartingPrice(
	shippingRates: CartShippingPackageShippingRate[]
): CartShippingPackageShippingRate | undefined {
	if ( shippingRates ) {
		return shippingRates.reduce(
			(
				lowestRate: CartShippingPackageShippingRate | undefined,
				currentRate: CartShippingPackageShippingRate
			) => {
				if ( currentRate.method_id !== 'local_pickup' ) {
					return lowestRate;
				}
				if (
					lowestRate === undefined ||
					currentRate.price < lowestRate.price
				) {
					return currentRate;
				}
				return lowestRate;
			},
			undefined
		);
	}
}
