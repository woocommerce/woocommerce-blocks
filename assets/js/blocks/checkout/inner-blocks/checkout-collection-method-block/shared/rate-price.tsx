/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { getSetting } from '@woocommerce/settings';
import { createInterpolateElement } from '@wordpress/element';
import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

export const RatePrice = ( {
	minRate,
	maxRate,
}: {
	minRate: CartShippingPackageShippingRate | undefined;
	maxRate: CartShippingPackageShippingRate | undefined;
} ) => {
	if ( ! minRate && ! maxRate ) {
		return null;
	}
	const minRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( minRate.price, 10 ) + parseInt( minRate.taxes, 10 )
		: parseInt( minRate.price, 10 );
	const maxRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( maxRate.price, 10 ) + parseInt( maxRate.taxes, 10 )
		: parseInt( maxRate.price, 10 );

	return (
		<span className="wc-block-checkout__collection-item-price">
			{ minRatePrice === 0 && maxRatePrice === 0 ? (
				<em>{ __( 'free', 'woo-gutenberg-products-block' ) }</em>
			) : (
				createInterpolateElement(
					__( 'from <price />', 'woo-gutenberg-products-block' ),
					{
						price:
							minRatePrice === 0 ? (
								<em>
									{ __(
										'free',
										'woo-gutenberg-products-block'
									) }
								</em>
							) : (
								<FormattedMonetaryAmount
									currency={ getCurrencyFromPriceResponse(
										minRate
									) }
									value={ minRatePrice }
								/>
							),
					}
				)
			) }
		</span>
	);
};
