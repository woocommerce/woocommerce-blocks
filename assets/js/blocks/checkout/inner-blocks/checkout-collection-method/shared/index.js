export { RatePrice } from './rate-price';
export * from './helpers';
