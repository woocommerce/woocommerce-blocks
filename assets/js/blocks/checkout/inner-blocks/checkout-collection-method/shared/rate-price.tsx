/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { getSetting } from '@woocommerce/settings';
import { createInterpolateElement } from '@wordpress/element';
import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

export const RatePrice = ( {
	rate,
}: {
	rate: CartShippingPackageShippingRate;
} ) => {
	const ratePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( rate.price, 10 ) + parseInt( rate.taxes, 10 )
		: parseInt( rate.price, 10 );
	return (
		<span className="wc-block-checkout__collection-item-price">
			{ ratePrice === 0
				? __( 'free', 'woo-gutenberg-products-block' )
				: createInterpolateElement(
						__( 'from <price />', 'woo-gutenberg-products-block' ),
						{
							price: (
								<FormattedMonetaryAmount
									currency={ getCurrencyFromPriceResponse(
										rate
									) }
									value={ ratePrice }
								/>
							),
						}
				  ) }
		</span>
	);
};
