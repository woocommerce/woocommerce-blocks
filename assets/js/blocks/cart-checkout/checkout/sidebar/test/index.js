/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { previewCart } from '@woocommerce/resource-previews';

/**
 * Internal dependencies
 */
import { allSettings } from '../../../../../settings/shared/settings-init';
import CheckoutSidebar from '../index';

describe( 'Testing checkout sidebar', () => {
	it( 'Shows rate percentages after tax lines if the block is set to do so', async () => {
		allSettings.displayCartPricesIncludingTax = false;
		allSettings.displayItemizedTaxes = true;
		const { totals: cartTotals, items: cartItems } = previewCart;
		const { container } = render(
			<CheckoutSidebar
				cartItems={ cartItems }
				cartTotals={ cartTotals }
				showRateAfterTaxName={ true }
			/>
		);
		expect( container ).toMatchSnapshot();
	} );
} );
