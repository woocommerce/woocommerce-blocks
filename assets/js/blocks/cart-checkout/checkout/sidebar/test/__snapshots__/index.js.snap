// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing checkout sidebar Shows rate percentages after tax lines if the block is set to do so 1`] = `
<div>
  <div
    class="wc-block-components-totals-wrapper"
  />
  <div
    class="wc-block-components-totals-wrapper"
  >
    <div
      class="wc-block-components-totals-item"
    >
      <span
        class="wc-block-components-totals-item__label"
      >
        Subtotal
      </span>
      <span
        class="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-components-totals-item__value"
      >
        $24.00
      </span>
      <div
        class="wc-block-components-totals-item__description"
      />
    </div>
  </div>
  <div
    class="wc-block-components-totals-wrapper"
  >
    <div
      class="wc-block-components-totals-coupon wc-block-components-panel"
    >
      <div>
        <button
          aria-expanded="false"
          class="wc-block-components-panel__button"
        >
          <svg
            aria-hidden="true"
            class="wc-block-components-panel__button-icon"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 9.4L12 14 7 9.4l-1 1.2 6 5.4 6-5.4z"
            />
          </svg>
          <span
            aria-hidden="true"
          >
            Coupon code
          </span>
          <span
            class="screen-reader-text"
          >
            Apply a coupon code
          </span>
        </button>
      </div>
      <div
        class="wc-block-components-panel__content"
        hidden=""
      >
        <div
          class=""
        >
          <div
            aria-hidden="false"
            class=""
          >
            <div
              class="wc-block-components-totals-coupon__content"
            >
              <form
                class="wc-block-components-totals-coupon__form"
              >
                <div
                  class="wc-block-components-text-input wc-block-components-totals-coupon__input is-active"
                >
                  <input
                    aria-describedby="wc-block-components-totals-coupon__input-0"
                    aria-label="Enter code"
                    autocapitalize="off"
                    autocomplete="off"
                    id="wc-block-components-totals-coupon__input-0"
                    type="text"
                    value=""
                  />
                  <label
                    for="wc-block-components-totals-coupon__input-0"
                  >
                    Enter code
                  </label>
                </div>
                <button
                  class="components-button wc-block-components-button wc-block-components-totals-coupon__button"
                  disabled=""
                  type="submit"
                >
                  <span
                    class="wc-block-components-button__text"
                  >
                    Apply
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="wc-block-components-totals-wrapper"
  >
    <div
      class="wc-block-components-totals-taxes"
    >
      <div
        class="wc-block-components-totals-item wc-block-components-totals-taxes__grouped-rate"
      >
        <span
          class="wc-block-components-totals-item__label"
        >
          Sales tax 20%
        </span>
        <span
          class="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-components-totals-item__value"
        >
          $6.00
        </span>
        <div
          class="wc-block-components-totals-item__description"
        />
      </div>
       
    </div>
  </div>
  <div
    class="wc-block-components-totals-wrapper"
  >
    <div
      class="wc-block-components-totals-item wc-block-components-totals-footer-item"
    >
      <span
        class="wc-block-components-totals-item__label"
      >
        Total
      </span>
      <span
        class="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-components-totals-item__value"
      >
        $30.00
      </span>
      <div
        class="wc-block-components-totals-item__description"
      />
    </div>
  </div>
</div>
`;
